src/components/editor/PanelMain.tsx

```diff
@@ -113,19 +134,26 @@ export const PanelMain: React.FC = () => {
         lastValue.current = newValue;
         setValue(newValue);
       }
+
+      if (metaframe.inputs.createdData) {
+        setLastChangedText(createdByText(metaframe.inputs.createdData))
+      }
     }
-  }, [metaframe.inputs, setValue, lastValue.current, valueName]);
+  }, [metaframe.inputs, lastValue.current, valueName]);
 
   useEffect(() => {
-    const inputs = metaframe?.metaframe?.getInputs();
+    // this should set the value only once so it doesn't clobber local state
+    if (!!localValue) return;
 
+    const inputs = metaframe?.metaframe?.getInputs();
     if (!inputs) {
       return;
     }
-    Object.keys(inputs).forEach((key) => {
```

Comment:
I reverted this, but this would overwrite the value saved to the editor if the input has multiple keys. Is that desired behavior?

Reply:
From our convo, if there are multiple keys, just use the first key, lexographically sorted.